app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,135 @@
+require 'pry'
+
+class RecipesController < ApplicationController
+  # before_action :authenticate_user!, except: [:show]
+
+  def index
+    @recipes = current_user.recipes
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def add_ingredient
+    @recipe = Recipe.joins(:recipe_foods).select('recipes.*,recipe_foods.quantity').find(params[:id])
+
+    if current_user == @recipe.user
+      @foods_not_in_recipe = Food.where.not(id: @recipe.foods.pluck(:id))
+
+      if params[:recipe].present? && params[:recipe][:food_ids].present?
+        food_ids = params[:recipe][:food_ids].reject(&:empty?) # Remove empty strings
+        Food.where(id: food_ids).each do |food|
+          RecipeFood.create!(food:, recipe: @recipe, quantity: params[:recipe][:quantity])
+        end
+        # @recipe.foods << foods_to_add
+        flash[:notice] = 'Ingredients added successfully.'
+        redirect_to @recipe
+        return
+      end
+    else
+      flash[:alert] = 'You do not have permission to add ingredients to this recipe.'
+    end
+
+    render 'add_ingredient'
+  end
```

Comment:
When attempting to add ingredients, I get the error shown below, which stops me from adding an ingredient. kindly do well to fix this as this can lead to a poor user experience👍🏼.
<img width="1256" alt="image" src="https://github.com/Katended/recipe-app/assets/64643995/a622ea0a-a354-4916-930c-f1260936bcf7">